sled-agent/health-monitor/src/health_checks.rs

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Helpers for running health checks from the sled agent
+
+use illumos_utils::svcs::Svcs;
+use illumos_utils::svcs::SvcsInMaintenanceResult;
+use slog::Logger;
+use tokio::sync::watch;
+
+pub(crate) async fn poll_smf_services_in_maintenance(
+    log: Logger,
+    smf_services_in_maintenance_tx: watch::Sender<
+        Result<SvcsInMaintenanceResult, String>,
+    >,
+) {
+    // We poll every minute to verify the health of all services. This interval
+    // is arbitrary.
+    let mut interval =
```

Comment:
I'd set the MissedTickBehavior to `Skip`.  If one of these invocations takes a while, that's not a reason to do a bunch in a row quickly.